temporal/server_option.go

```diff
@@ -126,10 +126,10 @@ func WithAudienceGetter(audienceGetter func(cfg *config.Config) authorization.JW
 //
 // Sample usage:
 // logger := log.NewCLILogger()
-// handler, err := NewCustomHandlerImplementation(logger)
-// reporter, err2 := metrics.NewEventsReporter(handler)
+// Provider, err := NewCustomProviderImplementation(logger)
+// reporter, err2 := metrics.NewEventsReporter(Provider)
 // server := temporal.NewServer(temporal.WithCustomMetricsReporter(repoter))
-// Deprecated use WithCustomMetricHandler
+// Deprecated use WithCustomMetricProvider
```

Comment:
Do we remove MetricHandler? How do user implement a custom metric consumer?

Reply:
Yes, we are removing MetricHandler and leaving just the MetricProvider.

This a very simple abstraction for implementing and is injected for consumption.